validator_client/src/duties_service.rs

```diff
@@ -179,17 +179,29 @@ impl SubscriptionSlots {
 
     /// Return `true` if we should send a subscription at `slot`.
     fn should_send_subscription_at(&self, slot: Slot) -> bool {
+        // Prevent subscriptions from being sent for duties which are in the past.
+        // If there are no subscription slots (should be impossible currently) then considering them
+        // expired is a safe fallback.
+        let expired = self
```

Comment:
Maybe we can log a warn (or have a metric) here for `expired == true`. If we are bundling this PR with #5305 , we should ideally be getting `should_send_subscription_at ` calls for expired slots only after a restart.
So getting a barrage of warns that are not right after a restart might alert us to a potential issue in duties calculation logic?